apps/aecore/lib/aecore/peers/worker.ex

```diff
@@ -189,10 +203,13 @@ defmodule Aecore.Peers.Worker do
       {:ok, info} ->
         case own_nonce == info.peer_nonce do
           false ->
-            if(info.genesis_block_hash == genesis_block_header_hash()) do
-              {:ok, info}
-            else
-              {:error, "Genesis header hash not valid"}
+            cond do
+              info.genesis_block_hash != genesis_block_header_hash() ->
+                {:error, "Genesis header hash not valid"}
+              !Map.has_key?(info, :server) || info.server != "aehttpserver"->
```

Comment:
Why actually we block peers that are not "aehttpserver"? If someone develops another AE client it's ok. As for validation if we are adding an AE node: we already check genesis_block_hash. If someone returns valid one they probably are an AE node.

Reply:
for now it is an easy check to filter other clients (e.g. a wallet) to try to connect